### PR TITLE
Heist Mode Improvements

### PR DESCRIPTION
**Pull request by CodenameB**
Everyone complains about heist mode being too fast. I have made this PR with the following suggested changes:

First, Raiders will now have multiple goals. Either two "easy" thefts, or one kidnap, and always one "hard" theft.

Second, with the theft goals split into two types, now raiders will need to plan a little better. Hard theft goals are things deep inside the station, or few in number. These include the complete particle accelerator, the nuke, one of the station's cats, a rubber ducky, a cyborg upload module, 4-6 subspace amplifiers (will require disassembling telecoms), the clown's mask, or human organs ("lungs" goal replaced with "appendix", because it didn't make sense to both try to minimize casualties and fatally wound people)

Easy objectives include: 4-6 guns, the supermatter shard, a jukebox, microwaves, cameras, plants, cyborg rechargers, fuel tanks, a plasma canister, and filing cabinets.

Objectives were chosen mostly based on their locations on Box, and to differentiate them from normal traitor objectives, plus humor, since who knows what goes on in the minds of Vox.

Third, I expanded the list of possible job roles that could be selected for kidnapping, and removed engineers. The previous list was almost exclusively jobs that worked near the edge of the station, making them easy to capture.

Finally, I added a 10 minute timer at round start to prevent the Vox from just heading home right away.

PLEASE REVIEW THIS VERY CAREFULLY. I have successfully compiled it, but I have not tested it, because I am not sure how.

:cl: 
 * rscadd: Heist mode tweaked. Now, the Vox Raiders will have one 'hard' theft goal, and then the possibility of a 'kidnap' goal or two 'easy' theft goals. 
* rscadd: There is also now a 10 minute timer that prevents Vox Raiders from instantly ending the round.

### CODE REVIEW

**Comment by ghost** [comment]
Codenameb for collab

**Comment by FudgePucker** [comment]
Cyborg upload is easy. You can bash and snatch the one in tech storage without ever leaving maintenance. Or print one from science.

**Comment by CodenameB** [comment]
Yes, but not without depressurizing maint and risking getting caught. Plus, any half-competent AI or security force will be watching secure storage like a hawk.

**Comment by FudgePucker** [comment]
Dude zero penalty mag boots LMAO.
Just park your ship near that maintenance and smash and grab. By the time they get magboots on to go after you, you are gone. It is currently listed as a hard objective so that means it will be a four min round.

**Comment by FudgePucker** [comment]
Also
>any half competent AI or security

What server do you play on because we are not on the same one.

**Comment by CodenameB** [comment]
> it will be a four min round.

Fudge, you clearly didn't read the rest of the post.

**Comment by TheFriggenSawbones** [comment]
god bless you I love you

**Comment by CptWad** [comment]
Theoretically all steal objectives are easy. Getting away with it is the hard part.

**Comment by CptWad** [comment]
Also
- seperate your changelog into parts for easy reading and comprehension.
- id go for 15 minutea

**Comment by SonixApache** [comment]
I don't think this addresses the issues with Raiders at all. 

It's a round-ending event like Ops, except these 'Ops' aren't really equipped or supposed to fight back with full force. Even if they do it 100% right they'll just jack the shit they need from under everyone's noses, and the round will end without anyone even knowing what happened.

Much like ye olde wizard, once bird ops are confirmed most folks with drop what they're doing to hunt the skipjack and end the round. Adding an arbitrary timelimit on that won't do much.

**Comment by FudgePucker** [comment]
Oh so the round will last 10 mins instead of four.

What

An

Improvement

File: code/game/gamemodes/heist/objectives.dm [hunk 1]
```diff
@@ -174,13 +281,13 @@
 				valid_organs++
 	return (valid_organs >= required_amount)
 
-/datum/theft_objective/number/heist/organs/lungs
+/datum/theft_objective/number/heist_hard/organs/appendix
 	name = "lungs"
```
Comment (CptWad) [inline code/game/gamemodes/heist/objectives.dm]: Change this to appendix too.

**Comment by CodenameB** [comment]
@SonixApache I've found the bigger problem is not valid hunters, but rather raiders rolling the "steal the supermatter shard" and getting a major victory before anyone has even left their departments for the first time. My goal was to force them to move around the station more, and ideally go inside. Almost all the time their two objectives will not be in the same place.

Is there a problem with validhunters going to the skipjack and forcing the round to end? Or can only the raiders end it?

@FudgePucker Perhaps you could suggest some better objectives instead of complaining?

**Comment by ghost** [comment]
Do they still have the dumb leave no vox behind?
they easily get red text Just because one box gets stunned and if they go back they risk dying too
Maybe add the "minimize casualties" objective TBH

**Comment by Kurfursten** [comment]
The problem with Heist is that the birds don't have the necessary gear to deal with a station of validhunters, or really even a couple of particularly nasty borgs.

I would give them a bunch of EMP gear (to disable lasers, borgs without being lethal), yellow gloves, night vision, and power sinks. Classic heist movie stuff - knock out power then go for your objectives. Plus, power outages are really noticable so the crew will know something is going down. Then, give the birds something sort of like a recall beacon, that they can activate on a short cooldown to warp home to the ship (to avoid a fight) or slap on an objective to warp it home.

Ideally the raiders would end up slippery like teleport wizards, but not as lethal.

**Comment by yclatious** [comment]
Make raiders infinite and be rewarded shit by hauling their theft of people and materials to the syndicate in exchange for telecrystals ot better gear.

Caw.

**Comment by Exxion** [comment]
The gloves raiders already get are insulated, but the rest sounds good

**Comment by IzayaChan** [comment]
>the complete particle accelerator

I didn't want vox raiders to win either.

**Comment by Shadowmech88** [comment]
Players have managed to succeed in stealing the PA before.

**Comment by CptWad** [comment]
PA objective is based on 

pa

**Comment by Exxion** [comment]
I was in a raider team that stole the PA completely undetected in under two minutes.
It's not hard, depending on how many engineers there are.

**Comment by CodenameB** [comment]
Stealing the PA is a little harder on Deff, and damn near impossible on Bagel.

**Comment by DamianX** [comment]
Why is the list of kidnap targets so restrictive? Can we add, like, most of the roles?

**Comment by Exxion** [comment]
That's a good point
The PA objective should probably be blacklisted on Bagel, or at least made to be the ONLY objective if they get it

Or, alternatively, make it an easy theft on all stations except Bagel and a hard one on Bagel

**Comment by SonixApache** [comment]
@CodenameB If you can steal a hand teleporter first it becomes a lot easier.

**Comment by CptWad** [comment]
Reminder that you can order a PA.
Its also given me an idea for bagels singlo but i digress

**Comment by CodenameB** [comment]
@DamianX The original list was even more restrictive. I think whoever wrote it had the thought of "steal people who would be useful to enslave" but I added some jobs who don't get to see much action, and are generally not on the outside edge of the station, like the bartender. Hopefully that makes their shift more interesting. We could certainly make it any job through.

**Comment by Shadowmech88** [comment]
I don't think it's a good idea to make it *any* job.
No one would care if an assistant was kidnapped, for example.

**Comment by gbasood** [comment]
Kidnap heads as three separate objectives in the same round soon(tm)

**Comment by FudgePucker** [comment]
Oh rubber ducky is also a "hard" objective.
One of the least visited places on the station is the public showers next to the holodeck, the second least visited areas is the holodeck.
Smash into though holodeck, grab from shower, leave though the way in. ZAS is your ally because zero down speed magboots.

What about a fireax as one of the hard objectives? Few obtainable, getting one from space would take hours, atmospherics is visited often and has major camera coverage even from space and is checked often, the other on station fireax is in the bridge for god and all to see.

**Comment by Shadowmech88** [comment]
I don't think the rubber ducky is a good objective. It's not rare or high-access, and there's only one, so aren't the raiders completely out of luck if some random assistant decides to pocket it?
They're going to have to turn the station upside-down to find it if it's not in its spawn location.

And if it *is* in its spawn location, it'll be too easy.

**Comment by CodenameB** [comment]
@FudgePucker You still don't seem to understand that causing a bunch of atmos alerts and depressurizing multiple rooms instantly blows your cover and is a terrible way to do it. Likewise, as Shadowmech points out, there is an extremely high chance that the ducky won't even be there. The fire axe is a less good choice, because there is one in atmos, unlocked with a simple multitool. It is the ducky problem, but many times worse because A: You only need to get through 2 rooms instead of 4, and B: No one ever takes it.

@Shadowmech88 Nope, there are actually two. There is the "easy" one that anyone can grab, and a second one in the Captain's Bathroom.

**Comment by FudgePucker** [comment]
If you save hard objectives for last you don't have to worry about blowing your cover if you are fast enough.

**Pull request merged**